framework/web/Request.php

```diff
@@ -196,6 +243,34 @@ public function resolve()
     }
 
     /**
+     * Filters headers according to the trusted host config.
+     * @param array $headers
+     * @since 2.0.13
+     * @return array filtered headers.
+     */
+    protected function filterHeaders(array $headers)
+    {
+        $host = $this->getUserHost();
```

Comment:
Hmm actually it should not use the user host at all. (Also that function doesn't exist).
It should be named `getRemoteHost()` (we're checking the next hop not the end user, the end user is never trusted to send us secure headers :p)

--> This means I need to double check the tests though; they should not have been passing.